### PR TITLE
feat: gw7812 get mentioned users from comment and send notification

### PR DESCRIPTION
**Pull request by mudana-grune**
https://youtrack.weseek.co.jp/issue/GW-7812
Implement method to get mentioned users from comment and send notification

## Testing Step
- Mention @username in comment editor and submit the comment (Eg: @mudana)
comment

- Login with mentioned username (@mudana), in-app notification will appear then click it
profile

- List of in-app notifications will appear
notification

- Test mention in a comment reply. Applied same steps as above

### CODE REVIEW

File: packages/app/src/server/models/comment.js [hunk 1]
```diff
@@ -94,6 +94,9 @@ module.exports = function(crowi) {
     return commentData;
   };
 
+  commentSchema.statics.findCommentById = async function(commentId) {
+    return this.findOne({ _id: commentId });
+  };
```
Comment (yuki-takei) [inline packages/app/src/server/models/comment.js:99]: This process is simple, so I think this method isn't needed. Please just call `findOne()` from CommentService directly.

**Review by yuki-takei — CHANGES_REQUESTED** [review]

File: packages/app/src/server/service/comment.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { stringifySnapshot } from '~/models/serializers/in-app-notification-snap
 import loggerFactory from '../../utils/logger';
 import Crowi from '../crowi';
 
+const USERNAME_PATTERN = new RegExp(/@[\w@.-]+/g);
```
[the remark this answers is not in the document]
Comment (mudana-grune) [reply packages/app/src/server/service/comment.ts]: Add regex example url

File: packages/app/src/server/models/comment.js (hunk 1 above)
Comment (mudana-grune) [reply packages/app/src/server/models/comment.js]: Removed method findCommentById and call findOne directly from CommentService

File: packages/app/src/server/models/user.js [hunk 3]
```diff
@@ -719,6 +719,10 @@ module.exports = function(crowi) {
     return this.find({ username: { $regex: username, $options: 'i' } }).limit(limit);
   };
 
+  userSchema.statics.findUserByUsernames = async function(usernames) {
```
[the remark this answers is not in the document]
Comment (mudana-grune) [reply packages/app/src/server/models/user.js]: Removed method findUserByUsernames and call find() directly from CommentService

**Pull request merged**